### PR TITLE
LB-1289: Fix XSPF playlist exporter bug and Enable XML support

### PR DESCRIPTION
**Pull request by yuktea**
# Problems

- The `serialize_xspf` method in `listenbrainz/webserver/views/playlist_api.py` has a bug which causes the `additional_metadata` field to be improperly constructed which leads to a crash.
- The response wasn't correctly formatted. 

# Solution
The following changes have been made:
- Fixed `serialize_xspf` to handle `additional_metadata` correctly. 
- Changed the response's content type in `get_playlist_xspf` to xml

### CODE REVIEW

File: listenbrainz/webserver/views/playlist_api.py [hunk 1]
```diff
@@ -143,11 +143,19 @@ def serialize_xspf(playlist: Playlist):
             playlist_extension_collaborator = ET.SubElement(playlist_extension_collaborators, "collaborator")
             playlist_extension_collaborator.text = collaborator
 
-    if playlist.additional_metadata:
-        playlist_extension_additional_metadata = ET.SubElement(playlist_extension, "additional_metadata")
-        for meta_key, meta_value in playlist.additional_metadata:
-            additional_metadata_item = ET.SubElement(playlist_extension_additional_metadata, "item", key=meta_key)
-            additional_metadata_item.text = meta_value
+        if playlist.additional_metadata:
```
Comment (amCap1712) [inline listenbrainz/webserver/views/playlist_api.py:146]: While testing the PR locally, the additional_metadata block wasn't showing up. Then I saw this entire block is over indented. It should be on the same indentation as other if blocks.

**Review by amCap1712 — CHANGES_REQUESTED** [review]
Thanks for the PR. I tested locally and it works. Let's add a test to ensure it remains that way.

**Comment by amCap1712** [comment]
While reviewing I also came across issue, LB-1489. Feel free to assign it yourself if you'd like to work on it next.

**Review by amCap1712 — APPROVED** [review]
LGTM, Thanks!

File: listenbrainz/tests/integration/test_playlist_api.py [hunk 2]
```diff
@@ -212,6 +212,35 @@ def test_playlist_create_empty(self):
...
+                    }
+                }
+            }
+        
+        response = self.client.post(
+            url_for("playlist_api_v1.create_playlist"),
+            json=playlist,
+            headers={"Authorization": "Token {}".format(self.user["auth_token"])}
+        )
+        self.assert200(response)
```
Comment (amCap1712) [inline listenbrainz/tests/integration/test_playlist_api.py:242]: You also need to make a call to the serialize_xspf endpoint after this, and ensure that it responds with a 200 response too.

Comment (yuktea) [reply listenbrainz/tests/integration/test_playlist_api.py:242]: Thanks for pointing that out

**Review by amCap1712 — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**